viewing_party/party.py

```diff
@@ -1,23 +1,128 @@
 # ------------- WAVE 1 --------------------
 
+from multiprocessing.sharedctypes import Value
+from re import I
+
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    
+    new_movie = {}
+    if title and genre and rating:
+        new_movie = {
+            "title" : title,
+            "genre" : genre,
+            "rating" : rating
+        }
+        return new_movie
+    return None
+
+def add_to_watched(user_data, movie):
+    user_data = {"watched" : [movie]}
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist" :[movie]}
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    average_rating = 0 
+    sum_list = []
+    for movie in user_data["watched"]:
+        if movie["rating"]:
+            sum_list.append(movie["rating"])
+        average_rating = sum(sum_list)/len(user_data["watched"])
+    return average_rating
 
+    
+def get_most_watched_genre(user_data):
+    genre_list =[]
+    for movie in user_data["watched"]:
+        if movie["genre"]:
+            genre_list.append(movie["genre"])
+            most_popular = max(set(genre_list), key = genre_list.count)
+        return most_popular
+    return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_friends_movies_list(user_data):
+    friend_list_movies = []
+    for my_dict in user_data["friends"]:
+        for friend_movie in my_dict["watched"]:

```

Comment:
`my_dict` isn't very descriptive. Other coders looking at this might not understand what kind of data it represents. Because it is a list of friends, maybe call it `friends`